flexmeasures/data/migrations/versions/a42b5124cc46_clean_sensors_to_show_from_asset_.py

```diff
@@ -0,0 +1,40 @@
+"""Remove sensors_to_show key from attributes column
+
+Revision ID: 2ba59c7c954e
+Revises: 950e23e3aa54
+Create Date: 2024-12-10 09:31:06.603743
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "2ba59c7c954e"
+down_revision = "950e23e3aa54"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # Fetch all rows in the generic_assets table, update 'attributes' column by removing the 'sensors_to_show' key
+    connection = op.get_bind()
+    table_name = "generic_asset"
+    attribute_key_to_remove = "sensors_to_show"
+
+    # Use raw SQL to fetch and update rows
+    connection.execute(
+        sa.text(
+            f"""
+        UPDATE {table_name}
+        SET attributes = attributes::jsonb - '{attribute_key_to_remove}'
+        WHERE attributes IS NOT NULL;
```

Comment:
Thanks. Does it also work if a row does have attributes (i.e. is not null) but the `"sensor_to_show"` key is not present (e.g. `attributes="{}"`)?

Reply:
Yes, that worked for me. Both `{}` or with existing attributes (which are now still there).

`attributes` is set to `NOT NULL` (I tried to test for that) so this case is not important.